apps/hpc-ftsadmin/src/app/components/filters/filter-pending-flows-table.tsx

```diff
@@ -0,0 +1,281 @@
+import { Form, Formik, FormikState } from 'formik';
+import { array, number, object, string } from 'yup';
+import tw from 'twin.macro';
+
+import { C } from '@unocha/hpc-ui';
+import {
+  FormObjectValue,
+  decodeFilters,
+  encodeFilters,
+} from '../../utils/parse-filters';
+import { t } from '../../../i18n';
+import { Query } from '../tables/table-utils';
+import { useContext } from 'react';
+import { AppContext } from '../../context';
+
+interface Props {
+  query: Query;
+  setQuery: (newQuery: Query) => void;
+  handleAbortController: () => void;
+}
+export interface PendingFlowsFilterValues {
+  status?: string;
+  dataProvider?: string;
+  reporterRefCode?: string;
+  sourceOrganizations?: Array<FormObjectValue>;
+  sourceCountries?: Array<FormObjectValue>;
+  destinationOrganizations?: Array<FormObjectValue>;
+  destinationCountries?: Array<FormObjectValue>;
+  destinationUsageYears?: Array<FormObjectValue>;
+  includeChildrenOfParkedFlows?: boolean;
+}
+
+export const PENDING_FLOWS_FILTER_INITIAL_VALUES: PendingFlowsFilterValues = {
+  status: '',
+  dataProvider: '',
+  reporterRefCode: '',
+  sourceOrganizations: [],
+  sourceCountries: [],
+  destinationOrganizations: [],
+  destinationCountries: [],
+  destinationUsageYears: [],
+  includeChildrenOfParkedFlows: false,
+};
+const StyledDiv = tw.div`
+my-6
+me-4
+lg:flex
+justify-end
+gap-x-4 
+`;
+export const FilterPendingFlowsTable = (props: Props) => {
+  const { setQuery, query, handleAbortController } = props;
+  const { lang, env } = useContext(AppContext);
+  const environment = env();
+
+  const FORM_VALIDATION = object().shape({
+    flowStatus: string(),
+    dataProvider: string(),
+    reporterRefCode: number()
+      .positive()
+      .typeError('Only positive integers are accepted'),
+    sourceOrganizations: array().of(
+      object().shape({ label: string(), id: string() })
+    ),
+    sourceCountries: array().of(
+      object().shape({ label: string(), id: string() })
+    ),
+    destinationOrganizations: array().of(
+      object().shape({ label: string(), id: string() })
+    ),
+    destinationCountries: array().of(
+      object().shape({ label: string(), id: string() })
+    ),
+    destinationUsageYears: array().of(
+      object().shape({ label: string(), id: string() })
+    ),
+  });
```

Comment:
By reading [official Formik documentation on validation](https://formik.org/docs/guides/validation), they have "baked-in support for schema-based form-level validation through Yup". Validation in Formik comes at form level and field level.

For form level validation, you can provide Yup codec via [`<Formik validationSchema>`](https://formik.org/docs/guides/validation#validationschema), but you can also use [`<Formik validate>`](https://formik.org/docs/guides/validation#validate) which accepts a method with `values` argument.

Since we're extensively using `io-ts` for runtime type checking, we can use it define codecs for usage inside `validate` method for form validation. You'll need to define custom codecs for things like positive integer, but there are examples of doing so in the codebase.